### PR TITLE
Fix flakey tests

### PR DESCRIPTION
**Pull request by AndyStabler**
- [ ] Clean up dangling processes
- [ ] Stop processes from dangling in the first place
- [x] Fix flakey timeout error

### CODE REVIEW

File: spec/tom_queue/worker_supervisor_spec.rb [hunk 1]
```diff
@@ -64,9 +64,9 @@
       supervisor.before_fork = -> { before_hook_message.set "executing before fork" }
 
       forked_supervisor.start
-
       expect(before_hook_message.wait).to eq "executing before fork"
-      forked_supervisor.term
+
+      forked_supervisor.kill # TODO: Why do we need a kill here to prevent orphan processes and nowhere else in the file?
```
Comment (AndyStabler) [inline spec/tom_queue/worker_supervisor_spec.rb:69]: Something to chat through @joshado . Adding `.kill` was the only way to stop the process being an orphan 🤷🏻

File: spec/tom_queue/tom_queue_integration_spec.rb [hunk 2]
```diff
@@ -342,7 +342,7 @@ def start!
       consumers.each do |c|
         total_size += c.work.size
         c.work.each do |work|
-          work.received_at.should < (work.run_at + 1.0)
+          work.received_at.should < (work.run_at + 2.0)
```
Comment (AndyStabler) [inline spec/tom_queue/tom_queue_integration_spec.rb:345]: "Fixes"  the occasional time out error

File: spec/tom_queue/deferred_work/deferred_work_manager_integration_spec.rb [hunk 3]
```diff
@@ -19,13 +19,15 @@ def notify(exception)
...
-    fork do
+    forked_process = TestForkedProcess.new do
       TomQueue.bunny = Bunny.new(TEST_AMQP_CONFIG)
       TomQueue.bunny.start
       TomQueue.exception_reporter = exception_reporter
       manager = TomQueue::DeferredWorkManager.new(TomQueue.default_prefix)
       manager.start
     end
+    forked_process.start
+    forked_process.pid
```
Comment (AndyStabler) [inline spec/tom_queue/deferred_work/deferred_work_manager_integration_spec.rb:30]: This means the orphan processes get cleaned up, but still need to figure out:

a) Why the tests fail
b) Why the processes are being orphaned

File: spec/tom_queue/tom_queue_integration_spec.rb (hunk 2 above)
Comment (caius) [reply spec/tom_queue/tom_queue_integration_spec.rb:345]: :car::dash: (👋🏻)

I think you can lean on the `be_within` matcher for this. `work.received_at.should be_within(2.0).of(work.run_at)`. (Might need to sprinkle `to_i` if those are time objects, can't remember if that compares nicely.)

Comment (AndyStabler) [reply spec/tom_queue/tom_queue_integration_spec.rb:345]: Oh hey there, @caius! Fancy seeing you here! `be_within` is much nicer 🎉  I'll get that changed up.

**Comment by AndyStabler** [comment]
Closing as @joshado and I (mainly @joshado) got the tests working/cleaned up directly in #68

**Pull request closed without merging**